File: object/object/src/Lec02/policy/discount/AmountDiscountPolicy.java
package Lec02.policy.discount;

import Lec02.condition.discount.discountCondition;
import Lec02.Money;
import Lec02.Screening;

public class AmountDiscountPolicy extends DefaultDiscountPolicy {


    private Money discountAmount;

    public AmountDiscountPolicy(Money discountAmount, discountCondition... conditions) {
        super(conditions);
        this.discountAmount = discountAmount;
    }

    @Override
    protected Money getDiscountAmount(Screening screening) {
        return discountAmount;
    }
}

File: object/object/src/java_to_kotlin/policy/discount/AmountDiscountPolicy.kt
package java_to_kotlin.policy.discount

import java_to_kotlin.Money
import java_to_kotlin.Screening
import java_to_kotlin.condition.discount.discountCondition

class AmountDiscountPolicy(private val discountAmount: Money, vararg conditions: discountCondition?) :
    DefaultDiscountPolicy(*conditions) {
    override fun getDiscountAmount(screening: Screening): Money? {
        return discountAmount
    }
}
